lua/luasnip/nodes/snippet.lua

```diff
@@ -20,6 +20,23 @@ local trig_engines = require("luasnip.nodes.util.trig_engines")
 local true_func = function()
 	return true
 end
+
+local default_context_matcher = function(self, line_to_cursor, match, captures)
+	-- backwards compatibility, try to invoke `condition`
+	if self.condition then
+		if not self.condition(line_to_cursor, match, captures) then
+			return nil
+		end
+	end
+
+	return {
+		trigger = match,
+		captures = captures,
+		trig_region = nil,
+		env_override = nil,
```

Comment:
WDYT about `trig_region` -> `clear_region`? To me, `trig_region` implies some interdependence with trigger, where `clear_region` is a bit more precise in what this is used for.

Reply:
i think `trig_region` is a property of this snippet expanding. it indicates the actual region where the snippets was triggered. The region is passed to `clear_region` because it needs to be cleared first in `Snippet:expand()`.

Reply:
I'm not sure I understand your response :/
I was suggesting to change the name from `trig_region` to `clear_region` (or something else :D) because `trig_region` sounds like it might have to be the range `trigger` occupies in the buffer, which is not the case.

Reply:
I mean that maybe the function's return should be more generic. The name `clear_region` sounds like this field is only be used in snippet expanding to clear triggered region. The `trig_region` means the actual region where the snippets was triggered. Maybe it can be used for other function.

Reply:
Ah alright, now I understand :+1:
Hm, for me the trigger is what you typed to invoke the snippet, and intuitively, I wouldn't say that the text that is matched with some condition belongs to it.
I'd even say that using `clear_region` to remove this text is a bit of a hack, but manually removing it in `context_matcher` will shift the expand-position and make `TM_CURRENT_LINE`, an env-variable which contains the line the snippet was expanded on, contain the wrong text (because the text that was there originally was removed), which means that snip_expand has to handle the removal after it saved the position and stored the current line

Reply:
emmm. Make sense! update it.